powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/database/repository/ApplicationRepository.java

```diff
@@ -20,11 +20,21 @@
 import io.getlime.security.powerauth.app.server.database.model.entity.ApplicationEntity;
 import org.springframework.data.repository.CrudRepository;
 
+import java.util.Optional;
+
 /**
  * Database repository class for access to applications
  *
  * @author Petr Dvorak, petr@wultra.com
  */
 public interface ApplicationRepository extends CrudRepository<ApplicationEntity, Long> {
 
+    /**
+     * Find application by name.
+     *
+     * @param applicationName Application name
+     * @return Optional application entity with given name, returns the first name match
+     */
+    Optional<ApplicationEntity> findFirstByName(String applicationName);
```

Comment:
As currently we do allow not unique application names I had to put here `findFirst` otherwise it could end up in non unique result. If we will decide to push our clients to ensure unique already stored application names we can use simple `findByName`.

Reply:
Yes, see my suggestion about updating existing DB values and adding the NOT NULL constraint.